src/transformers/pipelines/base.py

```diff
@@ -746,29 +746,33 @@ def _parse_and_tokenize(
         Parse arguments and tokenize
         """
         # Parse arguments
-        try:
-            inputs = self.tokenizer(
-                inputs,
-                add_special_tokens=add_special_tokens,
-                return_tensors=self.framework,
-                padding=padding,
-                truncation=truncation,
-            )
-        except ValueError:
-            # Can be linked to no padding token, if padding_token does not exist we should recover
-            inputs = self.tokenizer(
-                inputs,
-                add_special_tokens=add_special_tokens,
-                return_tensors=self.framework,
-                padding=False,
-                truncation=truncation,
-            )
-
+        inputs = self.tokenizer(
+            inputs,
+            add_special_tokens=add_special_tokens,
+            return_tensors=self.framework,
+            padding=padding,
+            truncation=truncation,
+        )
         return inputs
 
-    def __call__(self, *args, **kwargs):
-        inputs = self._parse_and_tokenize(*args, **kwargs)
-        return self._forward(inputs)
+    def __call__(self, inputs, *args, **kwargs):
+        try:
+            model_inputs = self._parse_and_tokenize(inputs, *args, **kwargs)
+            outputs = self._forward(model_inputs)
+            return outputs
+        except ValueError:
```

Comment:
Not of the scope of this PR, but more generally: I think we should have a proper `PaddingTokenMissingError` (or a better name) that we raise in this case, instead of relying on a generic `ValueError` that could come from something else.

Reply:
Agreed.